sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FlatMapGroupsWithStateExec.scala

```diff
@@ -250,6 +370,63 @@ case class FlatMapGroupsWithStateExec(
     }
   }
 
-  override protected def withNewChildInternal(newChild: SparkPlan): FlatMapGroupsWithStateExec =
-    copy(child = newChild)
+  override protected def withNewChildrenInternal(
+      newLeft: SparkPlan, newRight: SparkPlan): FlatMapGroupsWithStateExec =
+    copy(child = newLeft, initialState = newRight)
+}
+
+object FlatMapGroupsWithStateExec {
+
+  /**
+   * This method merges two grouped iterators such that we have a combined iterator
```

Comment:
it does not just combine.. it assumed the grouping iterator is sorted by key (first tuple in the input (InternalRow, Iterator[InterenalRow]) using the given comparator and then merges them according to that sorted order. you have to explain this bettr